src/api/common/schemas/amountbase.json

```diff
@@ -0,0 +1,44 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "amountbase",
+  "description": "Base class for amount and issue",
+  "type": "object",
+  "properties": {
+    "value": {
+      "description": "The quantity of the currency, denoted as a string to retain floating point precision",
+      "$ref": "value"
+    },
+    "currency": {
+      "description": "The three-character code or hex string used to denote currencies",
+      "$ref": "currency"
+    },
+    "counterparty": {
+      "description": "The Ripple account address of the currency's issuer or gateway, or an empty string if the currency is XRP",
+      "$ref": "address"
```

Comment:
The description says, "or an empty string if the currency is XRP" -- the schema just says it's not required, and additional properties are false. Doesn't that mean an empty string should fail validation when XRP is the currency? Or does it work anyway, since empty string is falsy in JS?